pennylane/_grad.py

```diff
@@ -288,9 +367,62 @@ def circuit(x, y):
 
     As expected, the tuple was unpacked and we directly received the Jacobian of the
     QNode with respect to ``y``.
+
+    We can also compute the Jacobian transformation inside a :func:`~.qjit` decorated program.
+
+    Consider this QNode
+
+    .. code-block::
+
+        dev = qml.device("lightning.qubit", wires=1)
+
+        @qml.qjit
+        def workflow(x):
+            @qml.qnode(dev)
+            def circuit(x):
+                qml.RX(np.pi * x[0], wires=0)
+                qml.RY(x[1], wires=0)
+                return qml.probs()
+
+            g = qml.jacobian(circuit)
+            return g(x)
+
+    >>> workflow(np.array([2.0, 1.0]))
+    array([[-1.32116540e-07,  1.33781874e-07],
+           [-4.20735506e-01,  4.20735506e-01]])
+
+    You can further compute the Jacobian transformation using other supported ``method``s
+    by :func:`catalyst.jacobian`.
+
+    .. code-block::
+
+        @qml.qjit
+        def workflow(x):
+            @qml.qnode(dev)
+            def circuit(x):
+                qml.RX(np.pi * x[0], wires=0)
+                qml.RY(x[1], wires=0)
+                return qml.probs()
+
+            g = qml.jacobian(circuit, method="fd", step_size=0.3)
+            return g(x)
+
+    >>> qml.qjit(workflow)(np.array([2.0, 1.0]))
+    array([[-0.37120096, -0.45467246],
+            [0.37120096,  0.45467246]])
     """
     # pylint: disable=no-value-for-parameter
 
+    if compiler.active("catalyst"):
```

Comment:
@josh146, regarding my comment in #4692, this is what I would very much like to avoid. How is the Jacobian operation supposed to know which compiler the user chose?

Reply:
I don't understand why this is now an issue :thinking:  We also discussed this in the ADR, and I don't remember it being a problem in the past.

Reply:
I appreciate the points everyone has made, but I'm a bit puzzled as to why this has become an issue within the context of our soft integration 🤔 We've talked about this in the ADR, and I can't recall it causing any problems in the past.

Reply:
I don't remember reading this in the ADR, could you point out where? As far as I remember the `active` function wasn't taking any arguments at all.